### DISCUSSION TITLE
Programmatically update form values?

### DISCUSSION THREAD

**Discussion by nico-martinucci** [TanStack/form#792] [Q&A] [answered]
For example, I might have a dropdown with some "pre-fill" options, each of which will fill some or all of the form values with some new default values, or a button to clear all values/restore to default, or an async function that updates a form with values from a pre-existing resource on load. Lots of use cases!

Is there a way/good pattern to do this?

**Comment by Balastrong** [comment, marked as answer]
For reference:
- [This](https://github.com/TanStack/form/discussions/728#discussioncomment-9893318) answers how to set a field manually.
- Reset with custom fields is being worked on, you can see more in #791 
- Getting default values from an API is covered in the [TanStack Query Integration example](https://tanstack.com/form/latest/docs/framework/react/examples/query-integration).
